libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -318,7 +318,7 @@ proc handleControl(g: GossipSub, peer: PubSubPeer, control: ControlMessage) =
 
 proc validateAndRelay(g: GossipSub,
                       msg: Message,
-                      msgId, msgIdSalted: MessageId,
+                      msgId: MessageId, msgIdSalted: SaltedId,
```

Comment:
why does `validationSeen` use `msgIdSalted`, but `mcache` uses `msgId`?

Reply:
the latter only holds validated data, therefore doesn't need to be salted - the former is controlled by a (potential) remote attacker (until it has passed validation).

Reply:
513ee0bba5b7713745632a6609c90693baf8357e